mmv1/templates/terraform/custom_flatten/vertex_ai_index_endpoint_private_service_connect_config.go.erb

```diff
@@ -0,0 +1,55 @@
+<%# The license inside this block applies to this file.
+	# Copyright 2023 Google Inc.
+	# Licensed under the Apache License, Version 2.0 (the "License");
+	# you may not use this file except in compliance with the License.
+	# You may obtain a copy of the License at
+	#
+	#     http://www.apache.org/licenses/LICENSE-2.0
+	#
+	# Unless required by applicable law or agreed to in writing, software
+	# distributed under the License is distributed on an "AS IS" BASIS,
+	# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	# See the License for the specific language governing permissions and
+	# limitations under the License.
+-%>
+func flatten<%= prefix -%><%= titlelize_property(property) -%>(v interface{}, d *schema.ResourceData, config *transport_tpg.Config) interface{} {
+    transformed := make(map[string]interface{})
+
+    if v == nil {
+        // Even when private_service_connect is set as false, API will not return object
```

Comment:
Sorry, just try to understand the API behavior. Does the API not return the object `private_service_connect` at all, even if we specify `project_allowlist` along with `enable_private_service_connect = false`? Does `project_allowlist` mean anything in this case (`enable_private_service_connect = false`) then?

Reply:
@shuyama1 Thank you for your review!

> Does the API not return the object private_service_connect at all, even if we specify project_allowlist along with enable_private_service_connect = false?

The API doesn't return any object at all. That's the case the latest `TestAccVertexAIIndexEndpoint_vertexAiIndexEndpointWithFalsePscExample` covers. I don't have permission to see the acceptance test log from [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8851/artifacts/35c17846-36ea-4465-86c5-f1956afa1ddd/recording/TestAccVertexAIIndexEndpoint_vertexAiIndexEndpointWithFalsePscExample.log)], but it should contain the GET request result after creating the test resource.

> Does project_allowlist mean anything in this case (enable_private_service_connect = false) then?

That's my understanding. `project_allowlist` doesn't mean anything when using `enable_private_service_connect = false`. 

---

The API accepted a POST request having `enable_private_service_connect = false` and no `network`, and the index endpoint was created. That said, I was wondering where the created index endpoint was accessible from. It might be accessible from nowhere. This is my speculation, though.

Reply:
@shotarok Thanks for the deep dive on this. I'll confirm with the API team and understand if that's the intended behavior and then see how we can represent it in Terraform. I'll update after I have an answer for them. Thank you and apologize for the delay on review.

Reply:
@shuyama1 Hello 👋 Just for checking in. Did you get any updates from the product team?

Reply:
@shotarok Sorry, have been busy weeks for the past few weeks with the major release and forgot to follow up with this PR. Big apologies! I've filed an issue internally and still waiting on responses. The most important thing I need to confirm with the product team is if 
```
"privateServiceConnectConfig": {
  "enablePrivateServiceConnect": false,
  "projectAllowlist": [
   "xxxxxxxx"
  ]
 }
```
is a valid config. And if it is, the API should return it back, as this will be hard to be handled by custom flattener on the provider end. 
I'll update it here as soon as I get back from them. Thanks for your patience.

Reply:
Thank you for your reply! I totally understand your concern. I'll wait for the update from the product team.

Reply:
@shuyama1 Hello, just for check-in 👋 Did you get an update from the product team?

Reply:
Sorry @shotarok for missing your comment. I haven't got an update yet but I'll try to ping them again today. No matter what I'll come up with a plan how we should move forward with this feature by tomorrow. Apologize for the waiting!

Reply:
Okay, I haven't got back from the service team but don't want to have you and customers waiting on this any more. Apologize for the wait. So here's the plan:
I don't think it makes sense to specify values for `project_allowlist` along with `enable_private_service_connect = false`, so let's just treat it as an invalid configuration (it's probably nice to have the service team add a validation on the backend though or definitely have the API return it if they think it's a valid case). So the only issue we need to resolve here is that `enable_private_service_connect = false` (nor the block `private_service_connect{}`) is not returned by the API even when we explicitly set it. Clearly, we still need to use the custom flattener here but to only write back `enablePrivateServiceConnect = false` when the block is not returned. No need to write back the `project_lllowlist` in this case and also I believe if we specify `enable_private_service_connect = true`, the API will return it so we also don't need to handle this case as well. A similar example I can find is [`approval_config.approval_required` in `cloudbuild_trigger`](https://github.com/hashicorp/terraform-provider-google-beta/blob/77533b2f16c343eaa058aba8678a8c374fb71a53/google-beta/services/cloudbuild/resource_cloudbuild_trigger.go#L120), and here is the custom_flattener for it: https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/templates/terraform/custom_flatten/cloudbuild_approval_required.go.erb

Reply:
Thank you for your detailed explanation! I updated the code in f3aecd4...74c92d6. Please let me know if the updated code is unexpected.